### PR TITLE
SA1028 (CodeMustNotContainTrailingWhitespace) cleanup

### LINKED ISSUE #798: Move SA1028 (NoTrailingWhitespace) documentation to repository
Currently the documentation for new rule SA1028 is located in the wiki. This file should be moved into the repository in the manner shown in #786, so users can more easily access it and pull requests can be submitted to update its content.

### LINKED ISSUE #799: Update documentation for SA1028 (NoTrailingWhitespace) to describe behavior
In #769, several questions were asked regarding the behavior of SA1028 in a variety of contexts. The answers to these questions should be included in the documentation for this rule (#798) so users and future developers are clear about the behavior without needing to look at the unit tests.

### LINKED ISSUE #800: SA1028 (NoTrailingWhitespace) metadata is inconsistent
The SA1028 analyzer does not follow several patterns that the other analyzers use:

* The type name does not match the title
* The category is not correct

### PR DESCRIPTION
**Pull request by sharwell**
Fixes #798 
Fixes #799 
Fixes #800

### CODE REVIEW

**Comment by vweijsters** [comment]
:+1: The changes look good to me

File: documentation/SA1028.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+| Disabled code (text located within a `#if`, `#elif`, or `#else` block which is not active) | |
+
+## How to fix violations
+
+The fix an instance of this violation, delete all excess whitespace characters at the end of the line.
+
+## How to suppress violations
+
+```csharp
+#pragma warning disable SA1028 // No trailing whitespace
```
Comment (AArnott) [inline documentation/SA1028.md]: Should the comments at the end of these `#pragma`'s be updated to match the new name?

**Comment by AArnott** [comment]
:+1:

File: documentation/SA1028.md (hunk 1 above)
Comment (sharwell) [inline documentation/SA1028.md]: They should, and this is exactly why I wanted this page in the repository :smile:

**Pull request merged**